e2e_tests/test_apps/easing.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+
+from textual.app import App, ComposeResult
+from textual.widgets import Placeholder
+
+
+class EasingApp(App):
+    css = """
+    Placeholder {
+        offset: 0 0;
+        transition: offset 500ms linear;
+    }
+    Placeholder.with-margin {
+        offset: 0 50%;
+        background: navy;
+    }
+    """
+
+    def on_load(self):
+        """Bind keys here."""
+        self.bind("q", "quit")
+
+    def compose(self) -> ComposeResult:
+        """Build layout here."""
+        yield Placeholder(name="Moving!", id="moving_placeholder")
+
+    def on_mount(self):
```

Comment:
I don't understand why this is working to be honest :sweat_smile: 
In the App class the implementation of `on_mount` is the following:
```py
    def on_mount(self) -> None:
        widgets = list(self.compose())
        if widgets:
            self.mount(*widgets)
            self.screen.refresh()
```

Fine, so it takes the widgets yielded by `self.compose()` and mount them.

However, if I override this `on_mount()` method like I do in this app, without calling `super().on_mount()`, I would expect to see this behaviour disappearing - and I shouldn't see the widgets defined in the `compose()` method, since it seems that the only place in the code where we're using `compose()` is in the `on_mount()` method?

And if I add `super.on_mount()` - which would have been why I was expecting to have to do on top of adding my own behaviours with this call to `self.set_timer()`, then the placeholder is added twice :thinking:

Reply:
Solved by asking Will: it turns out that Textual replicates a DOM-like events bubbling machinery, so even when we override the implementation of an event handler such as `on_mount` the parent implementation will still be called if the event's `prevent_default()` method is is not called :slightly_smiling_face: